Add countries include/exclude type filters

michaelwood opened the pull request.

Commit:
frontend: Convert country filters to include/exclude filters

Also tidy up some of the JS by looping over an array of the items when
the event setup is the same for each.

michaelwood commented:
@mariongalley - Just to double check before I do the wiring up of this:

This type of filter for best/grant location/recipient org countries:

<img width="376" height="713" alt="image" src="https://github.com/user-attachments/assets/2246f6b5-abac-4f1c-81cc-1dfb64f2fe0b" />

mariongalley commented:
@michaelwood That's the one, thanks!

mariongalley commented:
Can/should we do this for the other location filters too?

michaelwood commented:
> Can/should we do this for the other location filters too?

The main negative for this kind of filter is that it doesn't surface the information to the user until they start interacting with it (and it's a bit slow). 

We also have sufficient number of these widgets that after these 3 are implemented it will be needed to refactor the way they work to reduce the code complexity. Happy to do just will be a day needed instead of hours.

mariongalley commented:
@michaelwood Discussed with Ruth and agreed the other filters don't need it. The countries & regions filters has a small enough number of values that you could tick all but one by hand. The others don't really have a clear use case for all but one county or all but one LA.

Commit:
views: Number the facets/aggregates

The index number is used in multiple places so it's helpful to have this
in the documentation comment as well.

codemacabre reviewed (dismissed):
LGTM!

Commit:
views: correct spelling of recipient
